src/solvers/base_solver.cpp

```diff
@@ -175,7 +163,7 @@ void BaseSolver::InitializeOutput(const OutputType output_type, std::string root
       mesh_name << m_root_name << "-mesh." << std::setfill('0') << std::setw(6) << m_rank - 1;
       std::ofstream omesh(mesh_name.str().c_str());
       omesh.precision(8);
-      m_state[0].mesh->Print(omesh);
+      m_state.begin()->mesh->Print(omesh);
```

Comment:
It is strange to use begin() when you aren't going to iterate over the vector. Since this returns an iterator not a reference to the first item.  Mind changing this to front()?